### PR TITLE
Remove partially transferred songs 

### PR DESCRIPTION
**Pull request by reidbaker**
Ignore branch name. Also code review I may be :hammer:ed :)

### CODE REVIEW

File: SoundStream/src/com/lastcrusade/soundstream/manager/PlaylistDataManager.java [hunk 1]
```diff
@@ -124,6 +125,31 @@ private void pauseForNextRun() {
         } catch (InterruptedException e) {
         }
     }
+    
+    public void cleanRemotelyLoadedFiles(String disconnectedUserMac){
+        //Remove songs from mac that are in the process of transfering
+        Set<PlaylistEntry> toRemoveForRemoteLoad = new HashSet<PlaylistEntry>();
```
Comment (ejohnson44) [inline SoundStream/src/com/lastcrusade/soundstream/manager/PlaylistDataManager.java]: I don't really understand what this name means. I feel like the for doesn't fit

File: SoundStream/src/com/lastcrusade/soundstream/service/PlaylistService.java [hunk 2]
```diff
@@ -244,6 +246,20 @@ public void onReceiveAction(Context context, Intent intent) {
                 }
             }
         })
+        .addLocalAction(ConnectionService.ACTION_GUEST_DISCONNECTED, new IBroadcastActionHandler() {
+            @Override
+            public void onReceiveAction(Context context, Intent intent){
+                String guestMac = (String) intent.getExtras().get(ConnectionService.EXTRA_GUEST_ADDRESS);
+                mDataManager.cleanRemotelyLoadedFiles(guestMac);
+                Set<PlaylistEntry> toRemove = new HashSet<PlaylistEntry>();
```
Comment (theJenix) [inline SoundStream/src/com/lastcrusade/soundstream/service/PlaylistService.java]: Can we push this for loop/removeAll into the Playlist, into a method that removes songs for a given mac address?

**Comment by reidbaker** [comment]
My code cant compile unless I modify my .classpath. If someone else will run git checkout 8a164ef -- SoundStream/.classpath and tell me if that breaks your build please let me know.

**Comment by reidbaker** [comment]
The above comment shouldn't impact if this gets pulled in

**Comment by ejohnson44** [comment]
if @theJenix thinks everything is good then this can be pulled in. Looks good. :sheep:

**Comment by theJenix** [comment]
Two emphatic :thumbsup: A+++++ would review again.

**Pull request merged**